src/data/data.cc

```diff
@@ -210,6 +210,7 @@ DMatrix* DMatrix::Load(const std::string& uri,
     LOG(CONSOLE) << dmat->info().num_row << 'x' << dmat->info().num_col << " matrix with "
                  << dmat->info().num_nonzero << " entries loaded from " << uri;
   }
+  rabit::Allreduce<rabit::op::Max>(&dmat->info().num_col, 1);
```

Comment:
Can you add a motivating comment? Commit message might get lost over time, so having a comment explaining why this is needed will be useful.

Reply:
Yes, I can add a comment for this. What do you mean by a test to exploit the bug ? Reproduce the bug without the change ?

Reply:
Yep, a test which would ensure there's no regression.

Reply:
That's hard to reproduce. Dataset in libsvm with large number of features tend to have this problem. The one I am using is Criteo, which have 1 million features.